### PR TITLE
Job: #9110 Profiling BridgePoint 

### PR DESCRIPTION
**Pull request by travislondon**

### CODE REVIEW

File: doc-bridgepoint/notes/9110_Research_bp_profiling_tool/9110_Research_bp_profiling_tool.int.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+1. Abstract
+-----------
+This note describes the results from evaluation of commonly used java profilers against the BridgePoint tool.
+
+2. Document References
+----------------------
+<a id="2.1"></a>2.1 [BridgePoint DEI #9110](https://support.onefact.net/issues/9110) Research best tool for profiling BridgePoint and define profiling procedure  
+<a id="2.2"></a>2.2 [BridgePoint Profiler Research](https://docs.google.com/spreadsheets/d/1PILhtDHFWBIWIWv07Ik1mYs_8G_QLoGVgwZoQWTWcaQ/edit#gid=0) Initial research into profiling tools   
+<a id="2.3"></a>2.3 [BridgePoint Profiler](https://docs.google.com/document/d/1PdtBOeROEGpihp4jrgOM8JSUT57Ymsr9E8G7H2qATZ0/edit) Document describing tools further along with environment settings  
+
```
Comment (rmulvey) [inline doc-bridgepoint/notes/9110_Research_bp_profiling_tool/9110_Research_bp_profiling_tool.int.md]: Add a reference to jvisualvm (we need to know where to get it).

File: doc-bridgepoint/notes/9110_Research_bp_profiling_tool/9110_Research_bp_profiling_tool.int.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+3. Background
+-------------
+3.1 Performance issues with BridgePoint
+
+Notable performance issues have shown within the last year.  The tool needs to be profiled in order to determine the cause.  BridgePoint has never had a dedicated tool for profiling.  We need to determine which tool is best to suit the needs of BridgePoint development.
+
```
Comment (rmulvey) [inline doc-bridgepoint/notes/9110_Research_bp_profiling_tool/9110_Research_bp_profiling_tool.int.md]: Add a note about the suggestion provided by EclipseSource. What did they suggest and is they provided the "why" put it too?

File: doc-bridgepoint/notes/9110_Research_bp_profiling_tool/9110_Research_bp_profiling_tool.int.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+Suggestion:
+
+Use JProfiler.  This profiler provides more data than jvisualvm, but either would work.  JProfiler is preferred just from a UI/intuitive standpoint over YourKit.  Easy to use quickly and all the same data is present.
+
+After determining this suggestion this website was found.  The suggestion is in agreement:
+
+https://blog.oio.de/2014/03/07/java-profilers-a-short-comparison-between-jprofiler-yourkit-and-javas-visualvm/
+
+Procedures
+----------
```
Comment (rmulvey) [inline doc-bridgepoint/notes/9110_Research_bp_profiling_tool/9110_Research_bp_profiling_tool.int.md]: Number the sections below using our numbering convention (example 1.1.1.1)

File: doc-bridgepoint/notes/9110_Research_bp_profiling_tool/9110_Research_bp_profiling_tool.int.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  • Launch profiler   
+  • Launch BridgePoint   
+  • Connect profiler to BridgePoint process   
+  • Right before UI interaction begin profiling (example moving a shape)   
+  • Once the UI interaction is complete pause profiling   
+  • Go to Investigate profiling data step   
+  
+- Regression testing
+
+  • For any of the above profiling procedures after running them save a snapshot
```
Comment (rmulvey) [inline doc-bridgepoint/notes/9110_Research_bp_profiling_tool/9110_Research_bp_profiling_tool.int.md]: If a snapshot must be saved, put the step to do so in the procedure above. Also, we need a place to keep these snapshots. Where we put them we depend on their size. If they are large we will use s3, if they are not large we can use a standard spot we create in github. This procedure needs to define this location

**Comment by travislondon** [comment]
The requested changes have been made

**Comment by rmulvey** [comment]
The new procedure is being run (and thus tested) by Lee. I am waiting for feedback from this test to promote this work.

**Pull request merged**